app/src/main/java/com/babylon/wallet/android/data/repository/state/StateRepository.kt

```diff
@@ -313,20 +315,24 @@ class StateRepositoryImpl @Inject constructor(
         }
     }
 
-    override suspend fun getValidator(validatorAddress: String): Result<ValidatorDetail> = withContext(dispatcher) {
+    override suspend fun getValidator(validatorAddress: String): Result<ValidatorDetail> = getValidators(setOf(validatorAddress)).map {
+        it.first()
+    }
+
+    override suspend fun getValidators(validatorAddresses: Set<String>): Result<List<ValidatorDetail>> = withContext(dispatcher) {
         runCatching {
             val stateVersion = stateDao.getLatestStateVersion() ?: error("No cached state version found")
-            val validator = stateDao.getValidators(addresses = setOf(validatorAddress), atStateVersion = stateVersion).firstOrNull()
-            if (validator == null) {
+            val validators = stateDao.getValidators(addresses = validatorAddresses.toSet(), atStateVersion = stateVersion)
+            if (validators.size != validatorAddresses.size) {
```

Comment:
I would suggest here to check which validators are cached and which are not and only fetch the ones not cached. As we do in the rest of the functions. Since this is a set you can do that by

```
// validatorAddresses is already a Set
val cachedValidators = stateDao.getValidators(addresses = validatorAddresses.toSet(), atStateVersion = stateVersion)
val unkownAddresses = validatorAddresses subtract cachedValidators.map { it.address }
if (unknownAddresses.isNotEmpty()) {
   // fetch and store them
}

// Assemble cachedValidators and the fetched ones into the result
```

Reply:
sure, will do